Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -578,30 +588,73 @@ - (NSMutableOrderedSet*)mutableOrderedSetValueForKeyPath:(NSString*)keyPath {
     return StubReturn();
 }
 
-/**
- @Status Stub
- @Notes
-*/
-- (void)setNilValueForKey:(NSString*)key {
-    UNIMPLEMENTED();
+static SEL KVCValidatorForPropertyName(NSObject* self, const char* key) {
+    size_t len = strlen(key);
+    // For the key "example", we must construct the following buffer:
+    // _ _ _ _ _ _ _ _ _ x a m p l e _ _ _ _ _ _ _ \0
+    // and fill it with the following characters:
+    // v a l i d a t e E x a m p l e : e r r o r : \0
+    std::unique_ptr<char[]> chars(new char[8 + len + 8]{});
```

Comment:
avoid a heap allocation by using alloca... or just a locally-sized array. no need for this heap nonsense :)